drivers/w1/w1_ds2477_85_common.c

```diff
@@ -0,0 +1,294 @@
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	uint16_t delay_us = cfg->mode_timing[data->master_reg.od_active].t_slot * 8;
+	uint8_t rx_data;
+	int ret;
+
+	ret = cfg->w1_script_cmd(dev, delay_us, SCRIPT_OW_READ_BYTE, NULL, 0, &rx_data, 1);
+	if (ret != RESULT_SUCCESS) {
+		return -EIO;
+	}
+
+	return rx_data;
+}
+
+int ds2477_85_write_byte(const struct device *dev, const uint8_t tx_byte)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	uint16_t delay_us = cfg->mode_timing[data->master_reg.od_active].t_slot * 8;
+	uint8_t rx_data;
+	int ret;
+
+	ret = cfg->w1_script_cmd(dev, delay_us, SCRIPT_OW_WRITE_BYTE, &tx_byte, 1, &rx_data, 1);
+	if (ret != RESULT_SUCCESS) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int ds2477_85_write_block(const struct device *dev, const uint8_t *buffer, size_t tx_len)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	int w1_timing = cfg->mode_timing[data->master_reg.od_active].t_slot * 8 * tx_len;
+	uint8_t buf[3] = {CMD_WRITE_BLOCK, (tx_len + 1), 0};
+	struct i2c_msg tx_msg[2] = {
+		{.buf = buf, .len = 3, .flags = I2C_MSG_WRITE},
+		{.buf = (uint8_t *)buffer, .len = tx_len, .flags = (I2C_MSG_WRITE | I2C_MSG_STOP)}
+	};
+	int ret;
+
+	__ASSERT_NO_MSG(tx_len <= MAX_BLOCK_LEN);
+
+	if (tx_len == 0) {
+		return 0;
+	}
+
+	ret = i2c_transfer_dt(&cfg->i2c_spec, tx_msg, 2);
+	if (ret < 0) {
+		LOG_ERR("write block fail: %x", ret);
+		return -EIO;
+	}
+
+	k_usleep(cfg->t_op_us + (cfg->t_seq_us * tx_len) +  w1_timing);
+
+	ret = i2c_read_dt(&cfg->i2c_spec, buf, 2);
+	if (ret != 0 || buf[0] != 1 || buf[1] != RESULT_SUCCESS) {
+		LOG_ERR("invalid write block response: %x", ret);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int ds2477_85_read_block(const struct device *dev, uint8_t *buffer, size_t rx_len)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	int w1_timing = cfg->mode_timing[data->master_reg.od_active].t_slot * 8 * rx_len;
+	uint8_t buf[3] = {CMD_READ_BLOCK, 1, rx_len};
+	struct i2c_msg rx_msg[2] = {
+		{.buf = buf, .len = 2, .flags = I2C_MSG_READ},
+		{.buf = buffer, .len = rx_len, .flags = (I2C_MSG_READ | I2C_MSG_STOP)}
+	};
+	int ret;
+
+	__ASSERT_NO_MSG(rx_len <= MAX_BLOCK_LEN);
+
+	if (rx_len == 0) {
+		return 0;
+	}
+
+	ret = i2c_write_dt(&cfg->i2c_spec, buf, 3);
+	if (ret < 0) {
+		LOG_ERR("read block fail: %x", ret);
+		return -EIO;
+	}
+
+	k_usleep(cfg->t_op_us + (cfg->t_seq_us * rx_len) +  w1_timing);
+
+	ret = i2c_transfer_dt(&cfg->i2c_spec, rx_msg, 2);
+	if (ret != 0 || buf[1] != RESULT_SUCCESS) {
+		LOG_ERR("invalid read block response: %x", ret);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int ds2477_85_configure(const struct device *dev, enum w1_settings_type type, uint32_t value)
+{
+	struct w1_ds2477_85_data *data = dev->data;
+
+	switch (type) {
+	case W1_SETTING_SPEED:
```

Comment:
Is it expected to have mroe cases in the future? Otherwise I think if-else is more intuitive?

Reply:
Yes, there are definitely coming more, I just didn't include things like strong-pull ups yet because I'd like to check a few things first such that different master can do it in a uniform way.